### PR TITLE
Don't swallow errors from stackdriver init.

### PR DESCRIPTION
**Pull request by evankanderson**
Found this while poking at writing an e2e test case for stackdriver.

Without this patch, I get a panic. With the patch, I get:

> unable to init: stackdriver: couldn't initialize trace client: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.

/assign @jjzeng-seattle

### CODE REVIEW

File: metrics/stackdriver_exporter.go [hunk 1]
```diff
@@ -102,11 +102,12 @@ func init() {
...
-	if err == nil {
-		// Start the exporter.
-		// TODO(https://github.com/knative/pkg/issues/866): Move this to an interface.
-		e.StartMetricsExporter()
+	if err != nil {
+		return nil, err
 	}
+	// Start the exporter.
+	// TODO(https://github.com/knative/pkg/issues/866): Move this to an interface.
+	e.StartMetricsExporter()
```
Comment (vagababov) [inline metrics/stackdriver_exporter.go:110]: Well then we should consume the error from `StartMetricsExporter` as well.

Comment (evankanderson) [reply metrics/stackdriver_exporter.go]: Good catch, I didn't realize it returned an `error`.

**Pull request merged**